android/src/app/organicmaps/maplayer/MapButtonsViewModel.java

```diff
@@ -0,0 +1,75 @@
+package app.organicmaps.maplayer;
+
+import androidx.annotation.Nullable;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+public class MapButtonsViewModel extends ViewModel
+{
+  private final MutableLiveData<Boolean> mButtonsHidden = new MutableLiveData<>(false);
+  private final MutableLiveData<Float> mBottomButtonsHeight = new MutableLiveData<>(0f);
+  private final MutableLiveData<MapButtonsController.LayoutMode> mLayoutMode = new MutableLiveData<>(MapButtonsController.LayoutMode.regular);
+  private final MutableLiveData<Integer> mMyPositionMode = new MutableLiveData<>();
+  private final MutableLiveData<Mode> mMapLayerMode = new MutableLiveData<>();
+  private final MutableLiveData<SearchWheel.SearchOption> mSearchOption = new MutableLiveData<>();
```

Comment:
Here is the documentation: https://developer.android.com/topic/libraries/architecture/livedata#java.

**This is a lifecycle aware observable value**. When you create the view model, you tie it to an activity. Then the live data inside that view model follows the activity life cycle, so it does not trigger callbacks when the activity is stopped and automatically removes listeners when it is destroyed.

In our case, when we use the viewmodels in fragments we still have to remove the listeners because they have a shorter life cycle than the parent activity (to prevent the listeners from trying to update a destroyed UI).

Tying the view model to the parent activity allows us the reuse the value across different fragments. So the viewmodels store the UI state, and fragments can simply observe the live data to update their UI. No more `onSavedInstanceState`, no more dependency injection all over the place and we are sure all the fragments use the same state.

This makes debugging a lot easier and makes the codebase more robust. But then again I am no Android expert.